rsocket-examples/src/main/java/io/rsocket/examples/transport/ws/WebSocketHeadersSample.java

```diff
@@ -61,7 +63,10 @@ public static void main(String[] args) {
                         (in, out) -> {
                           if (in.headers().containsValue("Authorization", "test", true)) {
                             DuplexConnection connection =
-                                new WebsocketDuplexConnection((Connection) in);
+                                new ReassemblyDuplexConnection(
```

Comment:
Yeah... we always have to wrap every DuplexConnection like that to ensure correctness